screens/recipes/RecipesScreen.js

```diff
@@ -17,24 +17,37 @@ function Recipes(props) {
   const [errorMessage, setErrorMessage] = useState('');
   const deletedRecipe = useStoreState((state) => state.recipes.delete);
 
+  const loadRecipes = useStoreState((state) => state.recipes.load);
+  const setLoadRecipes = useStoreActions((actions) => actions.setLoadRecipes);
+
   useEffect(() => {
-    getRecipes()
-      .then((res) => {
-        setRecipes(res);
-      })
-      .catch((err) => {
-        setShowError(true);
-        setErrorMessage(err);
-      });
-  }, [newRecipe, getRecipes, deletedRecipe]);
+    if (loadRecipes) {
+      getRecipes()
+        .then((res) => {
+          setRecipes(res);
+        })
+        .catch((err) => {
+          setShowError(true);
+          setErrorMessage(err);
+        });
+      setLoadRecipes(false);
+      setLoadRecipes(false);
```

Comment:
Está duplicado el `setLoadRecipes`! Y metería ese set en un `finally`, que corre después de todos los `then`s y `catch`s:

```suggestion
    getRecipes()
      .then((res) => {
        setRecipes(res);
      })
      .catch((err) => {
        setShowError(true);
        setErrorMessage(err);
      })
      .finally(() => {
        setLoadRecipes(false);
      })
```

Así el estado `loadRecipes` solo será `false` cuando la request termina de verdad, cuéntame si te parece este cambio o si no se entiende!

Reply:
Está bien! el duplicado del `setLoadRecipes` se arregla en [#21](https://github.com/platanus/2021-1-Grupo4-Movil/pull/21)!

Reply:
Si! me gusta la opción de agregar el finally, no se me había ocurrido